### DISCUSSION TITLE
Saving issues on specific class after updating from 11.1 to 11.3.2

### DISCUSSION THREAD

**Discussion by dennSted** [pimcore/pimcore#17656] [Q&A / Support]
Hello,

as in #17641 described, I receive error messages when trying to save a specific class (the main product class). Further classes like e.g. Colours, Categories can be saved without any issues. Further, the error appears also when trying to save specific elements inside a created object using this class.

I'm not sure if has something to do with migrations issues or anything else. 

My error messages:
<img width="373" alt="saving_failed_1" src="https://github.com/user-attachments/assets/b01a9276-b107-4747-9749-515975081916">
<img width="558" alt="saving_failed_2" src="https://github.com/user-attachments/assets/470d51e4-4f59-4d6d-9154-b16adb11406f">

and inside the console:

`
Failed to load resource: the server responded with a status of 500 (Internal Server Error) - admin/class/save:1
`
`
xhr request to /admin/class/save failed - startup.js?_dc=15af3bbf97862be757d177e9d0954fb2dcabf4de
`

_I use different fields elements_:
Input, Label, Select, Image, Many-to-Many Object Relations, Multiselection, Many-to-one Relation, Localized fields, Objectbricks, Number and Table.

I hope anyone has an idea.

**Comment by DanielLaranjeira063** [comment]
Have you tried executing `bin/console p:d:classes-rebuild -d` ?
The -d option is used to remove classes that do not exist in the definitions anymore from the database.

**Reply by dennSted** [reply]
Hey,

thank you for the answer. Not before, but I've tried it right now and the error still exists. 

One thing I discovered (not proved to 100%), but only at Objects (using this class) created by a specific user this error also appears when trying to save the object.  The saving didn't work for this user as well as myself as admin user. The error message is a bit different, but from what I understand, it must be same reason...?

![image](https://github.com/user-attachments/assets/6071f1fd-77e6-4eae-acef-748e0cdf2748)

Editing the class didn't work eather.

**Comment by DanielLaranjeira063** [comment]
It seems that you have versions in the database that do not exist on the filesystem anymore.
If you do not need class versions, you could start fresh, by truncating the versions table and deleting everything inside the var/versions folder on your pimcore installation.

Other, but less likely option, is that it could be permissions issue on the var/versions folder,  you can also check if var/versions folder exists and has correct permissions.

**Reply by dennSted** [reply]
This worked for the error when trying to save an object. Something weird is that the permissions inside the object folder are different. So some directories where write-protected. After removing all versions from DB and file storage ([used this commands](https://pimcore.com/docs/platform/Pimcore/Administration_of_Pimcore/Cleanup_Data_Storage/#flush-all-versions-of-a-certain-type)) it's possible to save the existing objects. Some strange is that the object directory has now other permissions as the document/asset folder...
![image](https://github.com/user-attachments/assets/dde25013-4308-4557-997c-3c654d4528a1)
but the saving itself is working anyway.

Only the class itself inside Data Objects -> Classes can't be saved further (same error as in the start of this discussion).

**Comment by AlephLau** [comment]
@dennSted Try adding recursive permissions to the folder where your classes live, probably that's why it doesn't let you make changes to the file
`
sudo chmod 777 -R var/classes
`
